src/page/personal-information.page.ts

```diff
@@ -52,35 +53,30 @@ export class PersonalInformationPage {
   }
 
   private async selectContinent(desiredContinent: string): Promise<void> {
-    const selectedContinent = await this.continents.all(by.tagName('option')).filter(
-      async (elem) => {
-      const  continent = await elem.getText();
-      return continent === desiredContinent;
-    }).first();
+    const selectedContinent: ElementFinder = await this.continents.element(
+      by.cssContainingText('option', desiredContinent));
     await selectedContinent.click();
   }
 
   private async selectSeleniumComands(desiredComands: string[]): Promise<void> {
-    const selectedCommands: ElementFinder[] = await this.seleniumCommands.all(
-      by.tagName('option')).filter(async (elem) => {
-        const command = await elem.getText();
-        return desiredComands.includes(command);
-      });
-    selectedCommands.forEach(async (elem) => {
-      await elem.click();
+    desiredComands.forEach(async (comandName) => {
+      const comandObject: ElementFinder = await this.seleniumCommands.element(
+        by.cssContainingText('option', comandName)
+      );
+      await comandObject.click();
     });
   }
 
-  public async fillForm(fillValues): Promise<void> {
+  public async fillForm(fillValues: IPersonRegister): Promise<void> {
     this.dateInput;
     await this.firstNameInput.sendKeys(fillValues.firstName);
     await this.lastNameInput.sendKeys(fillValues.lastName);
     await this.selectRadioButton(fillValues.sex, this.sexOptions);
     await this.selectRadioButton(fillValues.experience.toString(), this.yearsOfExperienceOptions);
     await this.selectAllCheckBoxs(fillValues.profession, this.professionOptions);
     await this.selectAllCheckBoxs(fillValues.tools, this.seleniumTools);
-    await this.selectContinent(fillValues.continent);
     await this.selectSeleniumComands(fillValues.commands);
+    await this.selectContinent(fillValues.continent);
     await this.button.click();
```

Comment:
by doing an explicit wait for the button to be clickable, we don need to change the order of continent and commands, sorry for giving you a wrong advice

Reply:
Yes, that solved the problem :octocat: